### DISCUSSION TITLE
Performing Initial Adaptivity and user-object or post-processor?

### DISCUSSION THREAD

**Discussion by QuangDom** [idaholab/moose#20490] [Q&A General] [answered]
Hi,

I am wondering, is there any way to use the user-object or the post processor during the initial adaptivity? 
It seems too late to use it for the initial adaptivity, even though I already tried to set execute_on = 'INITIAL', but it doesn't help.

Thanks,
QD

**Comment by GiudGiud** [comment, marked as answer]
Hello

There is no earlier than INITIAL, so if it s not early enough then you may want to run adaptivity later.

Guillaume

**Reply by GiudGiud** [reply]
While it likely wont cover what you need for looking at who is being executed first (we're missing some granularity for this), two ways to examine the ordering of things are to:
- look at Moose.C https://github.com/idaholab/moose/blob/next/framework/src/base/Moose.C
- add this to the input file
```
[Debug]
  show_actions = true
[]
```
